tutorials/cloud-functions-async/index.md

```diff
@@ -0,0 +1,150 @@
+---
+title: Asynchronous patterns for Cloud Functions
+description: Build a simple HTTP function API for long running job or tasks.
+author: ptone
+tags: Cloud Functions, Firestore, serverless
+date_published: 2019-02-20
+---
+
+Preston Holmes | Solution Architect | Google
+
+<!-- diagram sources: https://docs.google.com/presentation/d/1s01eqo3YUKiskJwSESW-T17IeUQf3DLCT_lvuAV7CwM/edit#slide=id.g4fb0d7b3af_0_0 -->
+
+## Introduction
+
+This tutorial demonstrates how to use [Google Cloud Functions](https://cloud.google.com/functions/) to extend the synchronous web-hook style request/response to longer running jobs with a focus on trackable and stateful long running operations.
+
+When a caller makes a request of a service, it is asking it perform some work. Depending on the use-case, there are basically three expectations of this caller:
+
+### Synchronous Request/Response
+![sync](image/sync-request.png)
+
+The caller will wait until the work is done, expecting a result. This pattern can be directly and simply handled by [Cloud Functions](https://cloud.google.com/functions/) with an [HTTP trigger](https://cloud.google.com/functions/docs/calling/http).
+
+### Asynchronous Work Queue
+![work-queue](image/work-queue.png)
+
+The caller does not need to wait for the work to be done, and does not need to followup on the completion status.
+
+There are a couple ways this can be solved on Google Cloud Platform in a serverless way. You can use Cloud PubSub patterns for [long running tasks](https://cloud.google.com/solutions/using-cloud-pub-sub-long-running-tasks) or you can use a dedicated service with [Cloud Tasks](https://cloud.google.com/tasks/).
+
+### Asynchronous Stateful Jobs
+![](image/stateful-job.png)
+The caller does not want to wait for the work to be done, but would like to be able to inquire as to the completion of the request.
+
+This pattern is common enough that Google has defined a standard of a [long-running-operation API contract](https://github.com/googleapis/googleapis/tree/master/google/longrunning) used in multiple APIs.
+
+This is a basic pattern which may have some use-case specific details.
+
+## Serverless Stateful Jobs
+
+This tutorial uses several managed services to implement this pattern including [PubSub](https://cloud.google.com/pubsub/) and [Firestore](https://cloud.google.com/firestore/).
```

Comment:
Clarify what "this pattern" refers to. Serverless stateful jobs, I think, but should be explicit.

Also, reference to PubSub should be "Cloud PubSub", here and throughout.

Reply:
done